javascript/src/seedsource/containers/Upload.js

```diff
@@ -0,0 +1,152 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+import shp from 'shpjs'
+import { updateConstraintValues } from '../../actions/constraints'
+import { setError } from '../../actions/error'
+
+
+class Upload extends React.Component {
```

Comment:
Since this is a very specific-purpose upload, it's probably better to name this something like `ShapefileUpload` or `ShpUpload`.